### PR TITLE
extended view form for aggregate

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/state/src/recoil/distributions.ts [hunk 1]
```diff
@@ -35,11 +35,26 @@ import {
...
 import { State } from "./types";
+import { extendedSelection } from "./atoms";
+import { filters } from "./filters";
+import { groupSlice, groupStatistics } from "./groups";
 
 export type AggregationResponseFrom<
   TAggregate extends { response: { aggregate: readonly unknown[] } }
 > = TAggregate["response"]["aggregate"][0];
 
+const extendedViewForm = selector({
```
Comment (kaixi-wang) [inline app/packages/state/src/recoil/distributions.ts:46]: Does this have any relation to the StateForm.extended?

File: app/packages/state/src/recoil/distributions.ts [hunk 2]
```diff
@@ -35,11 +35,26 @@ import {
...
+  key: "extendedViewForm",
+  get: ({ get }) => {
+    return {
+      sampleIds: get(extendedSelection),
```
Comment (kaixi-wang) [inline app/packages/state/src/recoil/distributions.ts:50]: I'm making a selector for creating saved views that looks similar to this one, should we maybe make a form selector in `Selectors.ts` to reuse across the app? That way we don't have to check individual components when something is changed.
```
        send((session) =>
            saveView({
              onError,
              variables: {
                viewName: name,
                datasetName: datasetNameValue,
                viewStages: view,
                form: {
                  filters: snapshot.getLoadable(filters).contents,
                  sampleIds: [
                    ...snapshot.getLoadable(selectedSamples).contents,
                  ],
                  labels: snapshot.getLoadable(selectedLabelList).contents,
                  extended: snapshot.getLoadable(fos.extendedStages).contents,
                  slice:
                    snapshot.getLoadable(resolvedGroupSlice(false)).contents ||
                    null,
                },
                description,
                color,
                subscription,
                session,
              },

File: app/packages/relay/src/queries/countValues.ts [hunk 3]
```diff
@@ -1,11 +1,17 @@
 import { graphql } from "react-relay";
 
 export default graphql`
-  query countValuesQuery($dataset: String!, $view: BSONArray!, $path: String!) {
+  query countValuesQuery(
+    $dataset: String!
+    $view: BSONArray!
+    $path: String!
```
Comment (kaixi-wang) [inline app/packages/relay/src/queries/countValues.ts:7]: Probably a dumb question, but are the paths from the extended view not already calculated/handled)? I recall seeing some code in `core/aggregations.py` and `aggregations.ts` that seem to be constantly requerying aggregations, but I haven't looked into the exact parameters of the requests.
 
cc @benjaminpkane

**Review by kaixi-wang — APPROVED** [review]
lgtm and saw that you had already talked it over with ben 🚢

File: app/packages/state/src/recoil/distributions.ts (hunk 1 above)
Comment (imanjra) [reply app/packages/state/src/recoil/distributions.ts:46]: Not too sure if it is related to the `extended` property, but there is much overlap between StateForm and ExtendedViewForm.

File: app/packages/state/src/recoil/distributions.ts (hunk 2 above)
Comment (imanjra) [reply app/packages/state/src/recoil/distributions.ts:50]: Yeah, if this selector overlaps with the selector you're going to create then we can probably use a common selector for both.

Comment (imanjra) [reply app/packages/state/src/recoil/distributions.ts:50]: Currently, the aggregation form does not have `extended` but maybe we can a derived selector which is subset of above for form

File: app/packages/relay/src/queries/countValues.ts (hunk 3 above)
Comment (benjaminpkane) [reply app/packages/relay/src/queries/countValues.ts:7]: Previously aggregations were grouped into a single request. But now they are issued separately/explicitly as a parameter

File: app/packages/state/src/recoil/distributions.ts (hunk 2 above)
Comment (benjaminpkane) [reply app/packages/state/src/recoil/distributions.ts:50]: Yes, I would agree with a move to `selectors.ts` if you want to make that change @kaixi-wang

File: app/packages/state/src/recoil/distributions.ts (hunk 1 above)
Comment (benjaminpkane) [reply app/packages/state/src/recoil/distributions.ts:46]: They are definitely similar. They may be the exact same, but an audit would have to be done to be sure.

**Review by benjaminpkane — APPROVED** [review]
LGTM!

**Pull request merged**